Backpack/Badge/Classes/Badge.swift

```diff
@@ -0,0 +1,158 @@
+/*
+ * Backpack - Skyscanner's Design System
+ *
+ * Copyright 2018-2022 Skyscanner Ltd
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import UIKit
+
+@objcMembers
+@objc
+public class BPKBadge: UIView {
+    public var message: String? {
+        get { label.text }
+        set { label.text = newValue }
+    }
+    
+    public var type: BPKBadgeType = .success {
+        didSet { updateLookAndFeel() }
+    }
+    
+    public var icon: Icon? {
+        didSet {
+            updateLookAndFeel()
+        }
+    }
+    
+    private let label: BPKLabel = {
+        let label = BPKLabel(fontStyle: .textXsEmphasized)
+        label.translatesAutoresizingMaskIntoConstraints = false
+        return label
+    }()
+    private let iconView: UIImageView = {
+        let imageView = UIImageView(frame: .zero)
+        imageView.contentMode = .scaleAspectFit
+        imageView.translatesAutoresizingMaskIntoConstraints = false
+        return imageView
+    }()
+    private let containerStackView: UIStackView = {
+        let stack = UIStackView(frame: .zero)
+        stack.translatesAutoresizingMaskIntoConstraints = false
+        stack.axis = .horizontal
+        stack.spacing = BPKSpacingSm
+        return stack
+    }()
```

Comment:
Should we use `private lazy var` for these?

Reply:
I think this makes sense when the views are used further down the line, after the view's been initialised. This are all subviews that are used to setup the badge, so I don't think we win anything making them lazy. Also, it would force us to make them optional and to `?.` every time we want to use them.

Reply:
there's also the fact that making them `var`, even if they're private, they're communicating that are variables that are ok to be reassigned, and in this case does not make much sense... but I can agree that this is more on the philosophical side of the discussion hahaha